util/debug_test.go

```diff
@@ -177,3 +177,68 @@ func TestSchemaTreeString(t *testing.T) {
 		t.Errorf("got:\n%s\nwant:\n%s", got, want)
 	}
 }
+
+func TestDataSchemaTreesString(t *testing.T) {
+	containerWithListSchema := &yang.Entry{
+		Name: "container",
+		Kind: yang.DirectoryEntry,
+		Dir: map[string]*yang.Entry{
+			"key-list": {
+				Name:     "key-list",
+				Kind:     yang.DirectoryEntry,
+				ListAttr: &yang.ListAttr{MinElements: &yang.Value{Name: "0"}},
+				Key:      "key",
+				Config:   yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"key": {
+						Kind: yang.LeafEntry,
+						Name: "key",
+						Type: &yang.YangType{Kind: yang.Ystring},
+					},
+					"leaf-field": {
+						Kind: yang.LeafEntry,
+						Name: "leaf-field",
+						Type: &yang.YangType{Kind: yang.Yint32},
+					},
+					"leaf2-field": {
+						Kind: yang.LeafEntry,
+						Name: "leaf2-field",
+						Type: &yang.YangType{Kind: yang.Yint32},
+					},
+				},
+			},
+		},
+	}
+
+	type ListElemStruct struct {
+		Key        *string `path:"key"`
+		LeafField  *int32  `path:"leaf-field"`
+		Leaf2Field *int32  `path:"leaf2-field"`
+	}
+	type ContainerStruct struct {
+		KeyList map[string]*ListElemStruct `path:"key-list"`
+	}
+
+	container := &ContainerStruct{
+		KeyList: map[string]*ListElemStruct{
+			"keyval1": {
+				Key:       String("keyval1"),
+				LeafField: Int32(42),
+			},
+		},
+	}
+
+	got := DataSchemaTreesString(containerWithListSchema, container)
+	want := `     [container (container)]
+    KeyList [key-list (list)]
+    keyval1
+      Key : "keyval1" [key (leaf)]
+      LeafField : 42 [leaf-field (leaf)]
+`
+	if got != want {
```

Comment:
Consider using `pretty.Compare` or similar to generate a diff here? Seems like it might be easier to spot what the diff is in the case that the tests fail.

Reply:
For small protos I find dumping out the whole thing makes the diffs easier to spot. I agree though once it's more than a dozen lines or so, pretty's the way to go.